### ISSUE TITLE
Strange mail from YouTube, maybe related to this script?

### ISSUE THREAD

**Issue by Fabian42** [Elijas/auto-youtube-subscription-playlist-2#68]
I got mails from YouTube that asked me to fill in "YouTube API compliance audits". After contacting various semi-helpful Google support chats/phone numbers, I got a suspicion that this might be caused by this script.
Has anyone else experienced this and maybe even knows how to handle this? Apparently YouTube now imposed additional limits to my API usage. I didn't see any major issues so far from that, but maybe I will in the future.

**Comment by MrNgL** [comment]
I received the same emails as you, and I only have this script set-up on my account and it's the only one that uses YouTube API so I think it may be it. 
The first email I received was on February 13 which I decided to ignore, but on Feb 18 I got a reminder saying that I have 3 days to respond, failing which "my access to YouTube Data API would be affected".
Yesterday, Feb 25, I decided to fill out the Audit form where I explained that I use a script to automatically add videos to my playlists (I didn't link to this github project and never mentioned it).
I included a screenshot showing my gsheet (with blurred channels) and the app script code that uses the api, and explained that it's only for personal use.

I'll let you know if/when they'll reply me back.

**Comment by victorjzsun** [comment]
This might have to do with the fact that our script doesn't have a privacy policy, and thus is not verified by Google.

**Comment by Fabian42** [comment]
How did you fill out the form? It asks about details to a company, product, etc. and they also want you (/me) to give access to "my product" using login data etc. That is just impossible for me to do, especially since it's an addition to the form and there is zero information about where to send that.

**Comment by Fabian42** [comment]
Now they even said that they disabled my access to the YouTube API completely! If that is true, then I would be completely unable to use YouTube at all. I filled in only the form, saying "-" for most fields and "I have no company and I have no product, so stop harassing me about this!" in the first free-text field. @Elijas , please respond!

**Comment by Elijas** [comment]
Well, there should be no need for privacy policy, since we are not involved in product usage in any way.

I think that YouTube has caught on to our script and might see the API usage as abuse since we're not using API to integrate YouTube to Sheets, but rather use it as way enhance YouTube features themselves and that might be annoying to them for some reason, so they just want shut down this app and their users.

I don't know, just pure pessimistic speculation here l

**Comment by Fabian42** [comment]
I did the entire setup from the beginning again (which has the added advantage that I can use a version of the script that is not ancient) and this new spreadsheet seems to work. Let's see when Google starts to bug me about that one.

**Comment by MrNgL** [comment]
Update
Today, March 12, I received the response from the youtube team saying that they have completed the review and do not require any further actions from me at this time, but they may reach out again for a re-review. Everything is still working but I noticed that during these two weeks I received more "Summary of failures" notifications (Exceeded maximum execution time) than usual, might be unrelated to the process or maybe is due to the fact that I have close to 900 channels in the sheet (kept adding a few in those weeks).

Anyway here's how I filled the Quota Form if anyone else is interested:

**Reason for filling this form?**
Audit Initiated by YouTube 

**Your full legal name**
The name that I have on my acc

**Your organization's name**
none

**Your organization's website**
Link to the script (https://script.google.com/d/.......)

**Your organization's address**
none

**Organization contact email address**
The email address of the acc

**Describe your organization's work as it relates to YouTube**
I briefly explained that I use a script to add videos to my playlist, and how the script does the work

**Google representative email address**
left blank

**Content Owner ID (if available)**
left blank

**Please list all your API Client(s)**
subscriptionPlaylists Script 

**Please list all the project numbers used for each of your API Client(s)**
The Project number that you can find in your Google Cloud Console

**Is this a publicly or privately available API Client?**
Internal use only

**Please provide screenshots and instructions on how API Client accesses the YouTube Data API**
Again, a description of the script 

**If the screenshot is a file, please upload here.**
I uploaded a single screenshot showing part of the apps script code and part of my gsheet (with sensitive info censored)

**Where can we find each API Client(s)?**
Again the link to the script code (https://script.google.com/d/.......)

**If there is a log-in required to access the API client, please provide a demo account and instructions on how to access the API Client**
I explained to them that I don't know how to give them access to my google acc, even demo access

**Does your API Client commercialize YouTube Data?**
No

**Choose the option that best resembles your API Client's use case**
Other

**List other use cases for the API Client**
Oneliner explaining what the script does

**Specify all YouTube API Services used by this API Client**
I checked only Data API

**Select the primary audience for your API Client**
Strictly internal use only

**Approximately how many users use your API Client?**
1

**Explain how your API Client is used by your users**
Again, you know the drill, another description of the script. For some reason I made all those descriptions different from eachother, I think you can use always the same.

**Does your API Client use multiple projects to access YouTube APIs?**
No

**Does this API Client create, access or use any metrics derived from YouTube data?**
No

**Does this API Client display data from, or provide features or services across, multiple platforms (ex: Facebook, Twitter, etc.)?**
No

**Do you create/provide any type of reports using YouTube API Data?**
No

**How long do you store YouTube API Data?**
<24 hours

**How often do you refresh YouTube API Data?**
24 hours

**Does this API Client allow users to authenticate with their Google credentials**
No

**Send documents [...]**
Same screenshot as before


I'm sorry for the wall of text, I don't know why github doesn't have spoiler support in comments.

**Comment by Fabian42** [comment]
Are you sure that the reason why you got the mail is actually this script? They didn't tell me at all why they sent it to me.

**Comment by MrNgL** [comment]
> 
> 
> Are you sure that the reason why you got the mail is actually this script? They didn't tell me at all why they sent it to me.

Yes, it's the response to my submission of the form on Feb 25. I received the first confirmation immediately after filling out the quota form, and after more than 2 weeks I received the response review, I have attached a screenshot below.

![screenshot 2020-03-13 00 23 15](https://user-images.githubusercontent.com/20553443/76575975-0f840f00-64c1-11ea-982b-79b6ad592fc4.jpg)

**Comment by Fabian42** [comment]
I'm not sure what you answered there. My question was whether you know what caused you to receive the original mail from YouTube, the "action required" one. They don't name the project in there at all.

**Comment by MrNgL** [comment]
> 
> 
> I'm not sure what you answered there. My question was whether you know what caused you to receive the original mail from YouTube, the "action required" one. They don't name the project in there at all.

Oh I'm sorry, I though you were talking about the confirmation email I got after the audit.

The "Action Required" mail I got said "We are currently conducting a mandatory compliance review of your YouTube Data API Project." so I guess it's about the usage of the youtube API. At the end of the same email it says "For reference, your project number is ***" so they know which project is using their API, probably we started reaching the quota limit too often and we triggered something on their end, that led to all of this.

Here's a screen of the original mail I got:

![screenshot 2020-03-13 11 26 28](https://user-images.githubusercontent.com/20553443/76613082-bc8c7500-651d-11ea-8895-cbf97bf25162.jpg)

**Comment by Fabian42** [comment]
Thanks, I found it in my own mail now as well. But the ID is not the same as the one of the subscription playlist project. It's not even close to the same length.

**Comment by JesperDramsch** [comment]
I got it too and filled it out and it was fine.
